### PR TITLE
Python Wrapper for FHEcontext

### PR DESCRIPTION
**Pull request by rbharath**
This PR adds a Python wrapper for `FHEcontext`. This turned out to be much harder than expected due to many linker issues. Python's `distutils` support for building C/C++ extensions is really kludgy in places and not super documented. Managed to get it working on my laptop, but lots of hardcoded paths.

Marking WIP till travis passes.

### CODE REVIEW

**Comment by rbharath** [comment]
Build now functional on Linux but failing on Mac due to strange linker issues. Since none of us are developing on Mac, I'm going to just turn off the OSX build for time being. We can come back and fix this later.

File: setup.py [hunk 1]
```diff
@@ -4,37 +4,52 @@
...
-    The purpose of this class is to postpone importing pybind11
-    until it is actually installed, so that the ``get_include()``
-    method can be invoked. """
+  The purpose of this class is to postpone importing pybind11
+  until it is actually installed, so that the ``get_include()``
+  method can be invoked. """
 
-    def __init__(self, user=False):
-        self.user = user
+  def __init__(self, user=False):
```
Comment (robrobbins) [inline setup.py]: I dislike mismatched types when doing flags like this. `self.user` will likely be some object whenever it is eventually assigned yes? (or is it a string or some other primitive?). Thus having either an object (has been assigned) or a bool (unassigned) mismatch.

if its an object eventually use `None` as the falsy.

**Review by robrobbins — APPROVED** [review]
just the one comment about the user flag

File: setup.py (hunk 1 above)
Comment (rbharath) [reply setup.py]: Good comment. I copy/pasted this code from a PyBind11 tutorial, but we should clean this up a bit. I'll add this onto the clean-up list for the repo.

**Pull request merged**